docs/app/Examples/modules/Transition/Types/index.js

```diff
@@ -6,23 +6,30 @@ import ExampleSection from 'docs/app/Components/ComponentDoc/ExampleSection'
 
 const TransitionTypesExamples = () => (
   <ExampleSection title='Types'>
+
     <ComponentExample
       title='Transition'
       description='A Transition animates a single child by toggling the the visible prop.'
       examplePath='modules/Transition/Types/TransitionExampleTransition'
     >
       <Message warning>
-        <p>Do not unmount a Transition child or else it cannot be animated.</p>
+        <p>An animated component must not unmount unexpectedly or the transition will not run as intended.</p>
         <Message.List>
           <Message.Item>
-            Use the <code>unmountOnHide</code> prop to unmount the child after the animation exits.
+            If you need to animate children as they mount and unmount, use a <code>Transition.Group</code>. 
+          </Message.Item>
+          <Message.Item>
+            It is paramount that each child of a <code>Transition.Group</code>
+            has an unique <code>key</code> prop that stays constant through rerenders.
+            In particular, it is likely that <code>keys</code> based on pseudo-random identifiers or simple loop increment variables will lead to unexpected behavior.
```

Comment:
Per my comment, unique, consistent, and non-random keys are covered here:

https://facebook.github.io/react/docs/lists-and-keys.html#keys-must-only-be-unique-among-siblings

Reply:
I want to make sure I don't miss a nuance difference here.  @keksipurkki could you share what your keys were?  I currently don't see a difference between the Transition.Group keys and valid React keys.  Were they indexes by chance?  I could see this failing and possibly needing clarification.